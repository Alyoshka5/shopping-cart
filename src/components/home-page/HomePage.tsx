import '../../styles/HomePage.css';
import { Link } from 'react-router-dom';

export default function HomePage() {

    return (
        <div className="home-page">
            <img className='background-img' src="https://images.unsplash.com/photo-1518770660439-4636190af475?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1170&q=80" alt="" />
            <div className="content">
                <h1>Welcome to the <span className='company-name'>TechHub</span></h1>
                <div className='sub-text'>Discover a world of cutting-edge technology at our premier destination for a wide range of tech products. Explore the latest smartphones, laptops, audio devices, and smart home gadgets from top brands. Experience exceptional customer service, competitive prices, and embark on your tech journey with us today.</div>
                <Link to='/shop' className='call-to-action-button'>Shop Now</Link>
            </div>

        </div>
    )
}