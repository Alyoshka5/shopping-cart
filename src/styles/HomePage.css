.home-page {
    height: 100%;
    background-color: #000;
    margin: 0;
    padding: 0 5rem;
    display: flex;
    align-items: center;
    overflow: hidden;
}

.background-img {
    position: absolute;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100%;
    z-index: 1;
    opacity: 0.7;
}

.home-page .content {
    width: 60%;
    display: flex;
    flex-direction: column;
    gap: 2rem;
    z-index: 2;
}

.home-page h1 {
    font-size: 6rem;
    letter-spacing: 0.1rem;
    color: #fff;
    width: 80%;
    margin: 0;
}

.home-page h1 .company-name {
    color: var(--theme-color);
}

.home-page .sub-text {
    font-size: 1.5rem;
    color: #fff;
    width: 80%;
}

.call-to-action-button {
    cursor: pointer;
    font-size: 1.4rem;
    font-weight: 500;
    text-decoration: none;
    text-transform: uppercase;
    color: #000;
    background-color: var(--theme-color);
    width: fit-content;
    padding: 1rem 2rem;
    border: none;
    border-radius: 0.5rem;
}
.call-to-action-button:hover {
    background-color: var(--theme-color-dark);
}