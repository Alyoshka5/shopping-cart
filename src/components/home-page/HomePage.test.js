import rendrer from 'react-test-renderer';
import HomePage from './HomePage'
import { HashRouter } from 'react-router-dom';


it('renders home page content', () => {
    const pageTree = rendrer.create(<HashRouter><HomePage /></HashRouter>).toJSON();

    expect(pageTree).toMatchSnapshot();
})