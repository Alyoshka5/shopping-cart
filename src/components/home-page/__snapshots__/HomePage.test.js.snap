// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page content 1`] = `
<div
  className="home-page"
>
  <img
    alt=""
    className="background-img"
    src="https://images.unsplash.com/photo-1518770660439-4636190af475?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1170&q=80"
  />
  <div
    className="content"
  >
    <h1>
      Welcome to the 
      <span
        className="company-name"
      >
        TechHub
      </span>
    </h1>
    <div
      className="sub-text"
    >
      Discover a world of cutting-edge technology at our premier destination for a wide range of tech products. Explore the latest smartphones, laptops, audio devices, and smart home gadgets from top brands. Experience exceptional customer service, competitive prices, and embark on your tech journey with us today.
    </div>
    <a
      className="call-to-action-button"
      href="#/shop"
      onClick={[Function]}
    >
      Shop Now
    </a>
  </div>
</div>
`;
